### PR TITLE
Fix webdav mkdir for remote shares

### LINKED ISSUE #13546: Mkdir from sync client doesn't create folder on remote share
### Steps to reproduce
1. Create two users "user1" and "user2"
2. As "user1" create a folder "folder" with some pics in it
3. Share "folder" with link and allow editing
4. Login as "user2"
5. Mount the link as remote share (server to server sharing)
6. Setup the sync client for "user2"
7. On the local machine, create a folder "folder/sub"
8. Move all pics from "folder/*.jpg" to "sub"

### Expected result
Pics moved

### Actul result
415 Media type not supported
If you check "folder" as "user2" in the web UI, it seems that "sub" wasn't even created.

```
{"reqId":"8729a84a0ab3a589f7ca097deeb8c417","remoteAddr":"127.0.0.1","app":"webdav","message":"Exception: {\"Message\":\"No basic authentication headers were found\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/auth.php(123): Sabre\\\\DAV\\\\Auth\\\\Backend\\\\AbstractBasic->authenticate(Object(OC_Connector_Sabre_Server), 'ownCloud')\\n#1 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/auth.php(103): OC_Connector_Sabre_Auth->auth(Object(OC_Connector_Sabre_Server), 'ownCloud')\\n#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Auth\\\/Plugin.php(108): OC_Connector_Sabre_Auth->authenticate(Object(OC_Connector_Sabre_Server), 'ownCloud')\\n#3 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod('PROPFIND', '')\\n#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(433): call_user_func_array(Array, Array)\\n#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(455): Sabre\\\\DAV\\\\Server->broadcastEvent('beforeMethod', Array)\\n#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(214): Sabre\\\\DAV\\\\Server->invokeMethod('PROPFIND', '')\\n#7 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/files\\\/appinfo\\\/remote.php(61): Sabre\\\\DAV\\\\Server->exec()\\n#8 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/remote.php(54): require_once('\\\/srv\\\/www\\\/htdocs...')\\n#9 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Auth\\\/Backend\\\/AbstractBasic.php\",\"Line\":73}","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav"}
{"reqId":"b445fe7d210934b4f8b59c9f0441ccbb","remoteAddr":"127.0.0.1","app":"webdav","message":"Exception: {\"Message\":\"No basic authentication headers were found\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Auth\\\/Plugin.php(108): Sabre\\\\DAV\\\\Auth\\\\Backend\\\\AbstractBasic->authenticate(Object(OC_Connector_Sabre_Server), 'ownCloud')\\n#1 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod('PROPFIND', '')\\n#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(433): call_user_func_array(Array, Array)\\n#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(455): Sabre\\\\DAV\\\\Server->broadcastEvent('beforeMethod', Array)\\n#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(214): Sabre\\\\DAV\\\\Server->invokeMethod('PROPFIND', '')\\n#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/files_sharing\\\/publicwebdav.php(73): Sabre\\\\DAV\\\\Server->exec()\\n#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/public.php(48): require_once('\\\/srv\\\/www\\\/htdocs...')\\n#7 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Auth\\\/Backend\\\/AbstractBasic.php\",\"Line\":73}","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/public.php\/webdav\/"}
{"reqId":"ddc21404f40241ba431e42a4f6a1eabd","remoteAddr":"127.0.0.1","app":"webdav","message":"Exception: {\"Message\":\"No basic authentication headers were found\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Auth\\\/Plugin.php(108): Sabre\\\\DAV\\\\Auth\\\\Backend\\\\AbstractBasic->authenticate(Object(OC_Connector_Sabre_Server), 'ownCloud')\\n#1 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod('PROPFIND', '')\\n#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(433): call_user_func_array(Array, Array)\\n#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(455): Sabre\\\\DAV\\\\Server->broadcastEvent('beforeMethod', Array)\\n#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(214): Sabre\\\\DAV\\\\Server->invokeMethod('PROPFIND', '')\\n#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/files_sharing\\\/publicwebdav.php(73): Sabre\\\\DAV\\\\Server->exec()\\n#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/public.php(48): require_once('\\\/srv\\\/www\\\/htdocs...')\\n#7 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Auth\\\/Backend\\\/AbstractBasic.php\",\"Line\":73}","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/public.php\/webdav\/"}
{"reqId":"ae655c6735cdb5efe154a8df592c5529","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: storage_mtime at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/cache\/watcher.php#67","level":0,"time":"2015-01-21T12:32:52+00:00","method":"MKCOL","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2\/"}
{"reqId":"ae655c6735cdb5efe154a8df592c5529","remoteAddr":"127.0.0.1","app":"webdav","message":"Exception: {\"Message\":\"The resource you tried to create already exists\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(964): Sabre\\\\DAV\\\\Server->createCollection('folder\\\/sub2', Array, Array)\\n#1 [internal function]: Sabre\\\\DAV\\\\Server->httpMkcol('folder\\\/sub2')\\n#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(474): call_user_func(Array, 'folder\\\/sub2')\\n#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(214): Sabre\\\\DAV\\\\Server->invokeMethod('MKCOL', 'folder\\\/sub2')\\n#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/files\\\/appinfo\\\/remote.php(61): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/remote.php(54): require_once('\\\/srv\\\/www\\\/htdocs...')\\n#6 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php\",\"Line\":1714}","level":4,"time":"2015-01-21T12:32:52+00:00","method":"MKCOL","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2\/"}
{"reqId":"ae655c6735cdb5efe154a8df592c5529","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: mimetype at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/fileinfo.php#106","level":0,"time":"2015-01-21T12:32:52+00:00","method":"MKCOL","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2\/"}
{"reqId":"e94f317b9b81e4718179641ca21a9126","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: storage_mtime at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/cache\/watcher.php#67","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2"}
{"reqId":"d49fff7a69b68ba6f514fc60d358fdbb","remoteAddr":"127.0.0.1","app":"webdav","message":"Exception: {\"Message\":\"No basic authentication headers were found\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Auth\\\/Plugin.php(108): Sabre\\\\DAV\\\\Auth\\\\Backend\\\\AbstractBasic->authenticate(Object(OC_Connector_Sabre_Server), 'ownCloud')\\n#1 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod('PROPFIND', '')\\n#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(433): call_user_func_array(Array, Array)\\n#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(455): Sabre\\\\DAV\\\\Server->broadcastEvent('beforeMethod', Array)\\n#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(214): Sabre\\\\DAV\\\\Server->invokeMethod('PROPFIND', '')\\n#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/files_sharing\\\/publicwebdav.php(73): Sabre\\\\DAV\\\\Server->exec()\\n#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/public.php(48): require_once('\\\/srv\\\/www\\\/htdocs...')\\n#7 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Auth\\\/Backend\\\/AbstractBasic.php\",\"Line\":73}","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/public.php\/webdav\/"}
{"reqId":"e94f317b9b81e4718179641ca21a9126","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: mimetype at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/fileinfo.php#106","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2"}
{"reqId":"e94f317b9b81e4718179641ca21a9126","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: fileid at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/fileinfo.php#99","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2"}
{"reqId":"e94f317b9b81e4718179641ca21a9126","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: permissions at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/fileinfo.php#155","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2"}
{"reqId":"e94f317b9b81e4718179641ca21a9126","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: permissions at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/fileinfo.php#155","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2"}
{"reqId":"e94f317b9b81e4718179641ca21a9126","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: permissions at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/fileinfo.php#155","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2"}
{"reqId":"e94f317b9b81e4718179641ca21a9126","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: mimetype at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/fileinfo.php#106","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2"}
{"reqId":"e94f317b9b81e4718179641ca21a9126","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: permissions at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/fileinfo.php#155","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2"}
{"reqId":"e94f317b9b81e4718179641ca21a9126","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: etag at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/fileinfo.php#127","level":0,"time":"2015-01-21T12:32:52+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/folder\/sub2"}
```

(the auth header errors can be ignored, see #13545)

It seems something is wrong with "storage_mtime" and co.

Strangely if you create a folder over the web UI it will work properly.

@icewind1991

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/owncloud/core/issues/13546

@icewind1991 @DeepDiver1975 @MorrisJobke @nickvergessen please review

I'll also check whether this bug existed on stable7

### CODE REVIEW

**Comment by icewind1991** [comment]
Looks good :+1:

File: apps/files_sharing/tests/external/cache.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+class Test_Files_Sharing_External_Cache extends TestCase {
```
Comment (nickvergessen) [inline apps/files_sharing/tests/external/cache.php]: please use `namespace OCA\Files_sharing\Tests\External;` and `class Cache extends TestCase {`

Comment (PVince81) [inline apps/files_sharing/tests/external/cache.php]: I knew I shouldn't have summoned Mr Namespaces :wink: 

But thanks, that answers the question I had when I wrote it.

Comment (PVince81) [inline apps/files_sharing/tests/external/cache.php]: Done. Worked locally.

**Comment by MorrisJobke** [comment]
Works like a charm :+1:

**Comment by PVince81** [comment]
Seems to work fine on stable7, no backport needed.

**Pull request merged**